tools/isobuild/bundler.js

```diff
@@ -792,6 +806,33 @@ class Target {
                 this.nodeModulesDirectories[unibuild.nodeModulesPath] = nmd;
               }
               f.nodeModulesDirectory = nmd;
+
+              if (!archinfo.matches(this.arch, unibuild.arch)) {
+                // The unibuild we're trying to include doesn't work for the
+                // bundle target (eg, os.osx.x86_64 instead of os.linux.x86_64)!
+                // Hopefully this is because we specially enabled the feature
+                // that leads to this.
+                if (!this.providePackageJSONForUnavailableBinaryDeps) {
+                  throw Error("mismatched arch without special feature enabled "
```

Comment:
Running `git grep 'throw Error' tools` doesn't show me anything that looks at a skim like a user-facing error. If you disagree about any of them, I'd rather fix those cases to use buildmessage or a specific error type instead of adding 128 "this is an internal error" comments :)